src/bun.zig

```diff
@@ -3899,3 +3918,13 @@ comptime {
     // Must be nominal
     assert(GenericIndex(u32, opaque {}) != GenericIndex(u32, opaque {}));
 }
+
+/// Reverse of the slice index operator.
+/// Given `&slice[index] == item`, returns the `index` needed.
+/// The item must be in the slice.
+pub fn indexOfPointerInSlice(comptime T: type, slice: []const T, item: *const T) usize {
```

Comment:
does not account for alignment.

Reply:
doesnt sizeOf return the size of the item in the slice?

if so, this function asserts the item is in the slice; regardless of how the slice is aligned, the item must be at some multiple of the size after the pointers are subtracted.